internal/security/sensitive_string.go

```diff
@@ -0,0 +1,54 @@
+package security
+
+import (
+	"encoding/json"
+)
+
+const redactionPlaceholder = "***"
+
+// SensitiveString is a type for handling sensitive information securely.
+// This helps to achieve the following goals:
+//  1. Prevent accidental logging of sensitive information.
+//  2. Provide controlled visibility (e.g., redacted output for String() or MarshalJSON()).
+//  3. Enable controlled access to the sensitive value when needed.
+type SensitiveString struct {
+	value string
```

Comment:
@osmaczko Hmm, I'm not sure if this is right. I don't recall any other place we stored a string by pointer 🤔 
If one would want to avoid copy, they could pass `*SensitiveString`, no?

Reply:
I was worried about copy on `SensitiveString` creation. I've just learned that "[A string is represented in memory as a 2-word structure containing a pointer to the string data and a length](https://research.swtch.com/godata).", meaning the actual string data is not copied.